Implement route export

dandeduck opened the pull request:
It's purpose is to create the URI from which we can create the intent that will lead the user to his nav app of choice.
A couple of things:
1. I used the link to google maps, but android will ask the user to pick an app still.
2. I had to add some more stuff to objects (mostly to location) some of them related to this PR (address field) and some (LatLng) related to the upon arrival stuff.

dandeduck commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Export route to whatever Waze or Google maps needs](https://trello.com/c/JLu6hFJq/53-export-route-to-whatever-waze-or-google-maps-needs)

tomtzook commented on app/src/main/java/com/navapp/navigation/route/RouteLinkFactory.java:
```diff
@@ -0,0 +1,68 @@
+    }
+
+    private static String originSection(Route route) {
+        return new StringBuilder()
```
It would be better to use the same `StringBuilder` as from the original caller instead.

dandeduck replied:
yeah ok

tomtzook commented on app/src/main/java/com/navapp/navigation/route/Route.java:
```diff
@@ -17,15 +18,23 @@ public Route(List<Destination> destinations) {
+    }
+
+    public List<Destination> getWaypoints() {
+        return new ArrayList<>(destinations.subList(1, destinations.size()-1));
```
Why not use `subList` directly?

dandeduck replied:
hmm

tomtzook commented on app/src/test/java/com/nvapp/navigation/route/RouteLinkFactoryTest.java:
```diff
@@ -0,0 +1,62 @@
+    private String baseExpectedLinkText;
+
+    @Before
+    public void createRoute() {
```
Do https://www.testwithspring.com/lesson/writing-parameterized-tests-with-junit-4/

dandeduck replied:
can you explain how should I parameterize it? and why?

dandeduck replied:
I mean, why isn't this good enough?

dandeduck replied:
by explain, which part should I parameterize? (besides everything)

dandeduck replied:
will add this to route link builder test

tomtzook reviewed (changes requested):
Do stuff

dandeduck commented:
In addition to fixing everything besides the tests, I changed the api used for lat long and stuff (there is a legit reason for it)  and renamed location to `Place`. Should also rename `Destination` because in maps api it means the end point of a route.

tomtzook commented on app/src/main/java/com/navapp/navigation/destination/data/Place.java:
```diff
@@ -0,0 +1,65 @@
+package com.navapp.navigation.destination.data;
+
+import com.google.maps.*;
```
bad boy

dandeduck replied:
Forgot to change this setting

tomtzook commented on app/src/main/java/com/navapp/navigation/destination/PlaceDetailsFactory.java:
```diff
@@ -0,0 +1,20 @@
+package com.navapp.navigation.destination;
+
+import com.google.maps.*;
```
bad boy

dandeduck commented:
You didn't reply to the stuff from prev cr
